### PR TITLE
Markdown Article Support

### PR DESCRIPTION
**Pull request by Nomandes**
# Markdown Article Support

The static pages for terms of service, privacy and impressum are now created by parsing a local markdown file. Additionally the theming was tweaked to incorporate dynamic font sizes and improve readability.

### CODE REVIEW

File: packages/frontend/src/common/components/MarkdownArticle.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    { markdownComponent: "h3", muiTypoVariant: "h3" },
+    { markdownComponent: "h4", muiTypoVariant: "h4" },
+    { markdownComponent: "h5", muiTypoVariant: "h5" },
+    { markdownComponent: "h6", muiTypoVariant: "h6" },
+    { markdownComponent: "p", muiTypoVariant: "body1" },
+  ];
+
+  const components: { [key: string]: React.ComponentType<PropsWithChildren> } = {};
+
+  componentMappings.reduce((acc, { markdownComponent, muiTypoVariant }) => {
```
Comment (Nomandes) [inline packages/frontend/src/common/components/MarkdownArticle.tsx:28]: Not sure if this should kept simple but redundant after the KISS principle as it was implemented in the last commit

File: packages/frontend/src/common/context/ColorThemeContext.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React, { useState } from "react";
-import { createTheme, Theme } from "@mui/material";
+import { createTheme, responsiveFontSizes, Theme } from "@mui/material";
 import { ThemeProvider } from "@mui/material/styles";
 
 interface Props {
```
Comment (NearW) [inline packages/frontend/src/common/context/ColorThemeContext.tsx:5]: replace with PropsWithChildren

File: packages/frontend/src/common/context/ColorThemeContext.tsx [hunk 3]
```diff
@@ -58,6 +113,7 @@ const darkTheme = createTheme({
     highlightColorPrimary: "#b41964",
     highlightColorSecondary: "#14b400",
   },
+  ...typography,
 });
 
 export const ColorThemeContext = React.createContext<ColorThemeContextValues>(undefined!);
```
Comment (NearW) [inline packages/frontend/src/common/context/ColorThemeContext.tsx:119]: Rename to ThemeContext

File: packages/frontend/src/common/context/ColorThemeContext.tsx [hunk 4]
```diff
@@ -22,8 +30,53 @@ declare module "@mui/material/styles" {
...
+      defaultProps: {
+        variantMapping: {
+          title: "h1",
+          menuTitle: "span",
+        },
+      },
+    },
+  },
+};
+const typography = {
```
Comment (NearW) [inline packages/frontend/src/common/context/ColorThemeContext.tsx:57]: find type. Maybe extend TypographyVariantsOptions from existing typography variants?

File: packages/frontend/src/common/context/ColorThemeContext.tsx [hunk 5]
```diff
@@ -22,8 +30,53 @@ declare module "@mui/material/styles" {
...
+    menuTitle: React.CSSProperties;
+  }
+
+  // allow configuration using `createTheme`
+  interface TypographyVariantsOptions {
+    title?: React.CSSProperties;
+    menuTitle: React.CSSProperties;
+  }
+}
+const components = {
```
Comment (NearW) [inline packages/frontend/src/common/context/ColorThemeContext.tsx:45]: find type here

File: packages/frontend/src/common/components/MarkdownArticle.tsx (hunk 1 above)
Comment (Nomandes) [reply packages/frontend/src/common/components/MarkdownArticle.tsx]: We reworked the reduce to a mapping. So this is solved

File: packages/frontend/src/common/context/ColorThemeContext.tsx (hunk 4 above)
Comment (Nomandes) [reply packages/frontend/src/common/context/ColorThemeContext.tsx:57]: Were able to set it to a subtype of the VariantOptions

File: packages/frontend/src/common/context/ColorThemeContext.tsx (hunk 5 above)
Comment (Nomandes) [reply packages/frontend/src/common/context/ColorThemeContext.tsx:45]: Were able to set it to a subtype of the VariantOptions

File: packages/frontend/src/common/context/ThemeContext.tsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
+import React, { PropsWithChildren, useState } from "react";
+import { createTheme, responsiveFontSizes, Theme, ThemeOptions } from "@mui/material";
+import { ThemeProvider } from "@mui/material/styles";
+
+export interface ColorThemeContextValues {
```
Comment (NearW) [inline packages/frontend/src/common/context/ThemeContext.tsx:5]: Rename `ThemeContexstValues`

File: packages/frontend/src/common/context/ThemeContext.tsx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+  currentTheme: Theme;
+  setDarkTheme: () => void;
+  setLightTheme: () => void;
+}
+
+declare module "@mui/material/styles" {
```
Comment (NearW) [inline packages/frontend/src/common/context/ThemeContext.tsx:11]: Move declared modules to own file `mui.types.d.ts`

File: packages/frontend/src/common/context/ThemeContext.tsx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    background: {
+      default: "#B0BEC5",
+      paper: "#CFD8DC",
+    },
+    highlightColorPrimary: "#f05a96",
+    highlightColorSecondary: "#82c864",
+  },
+  ...typography,
+  ...components,
+});
```
Comment (NearW) [inline packages/frontend/src/common/context/ThemeContext.tsx:143]: remove. please add a eslint rule, that console.logs should not be allowed

**Review by NearW — APPROVED** [review]

**Pull request merged**